packages/calcite-components/src/components/block/block.tsx

```diff
@@ -237,29 +237,31 @@ export class Block
   }
 
   renderIcon(): VNode[] {
-    const { el, loading, messages, status } = this;
-
-    const statusIcon = ICONS[status];
-
-    const hasIcon = getSlotted(el, SLOTS.icon) || statusIcon;
-
-    const iconEl = !statusIcon ? (
-      <slot key="icon-slot" name={SLOTS.icon} />
-    ) : loading ? (
-      <calcite-loader inline label={messages.loading} />
-    ) : (
-      <calcite-icon
-        class={{
-          [CSS.statusIcon]: true,
-          [CSS.valid]: status == "valid",
-          [CSS.invalid]: status == "invalid"
-        }}
-        icon={statusIcon}
-        scale="m"
-      />
-    );
+    const { loading, messages, status } = this;
+
+    const hasSlottedIcon = !!getSlotted(this.el, SLOTS.icon);
 
-    return hasIcon ? <div class={CSS.icon}>{iconEl}</div> : null;
+    return loading ? (
+      <div class={CSS.icon}>
```

Comment:
Looks like the conditionally rendered container for the icon is the same and could be reused across renders. Could you set `key` on `<div class={CSS.icon}>` for this?